### PR TITLE
fix gnn shooting edge case

### PR DESCRIPTION
**Pull request by tomsilver**

### CODE REVIEW

File: tests/approaches/test_gnn_policy_approach.py [hunk 1]
```diff
@@ -200,3 +200,5 @@ def _solved_classifier(s, o):
                                         test_task.goal_holds,
                                         max_num_steps=CFG.horizon)
     assert "Option plan exhausted" in str(e)
+    # TODO: test that shooting does not infinitely hang in the case where the
```
Comment (ronuchit) [inline tests/approaches/test_gnn_policy_approach.py:203]: `git grep "MockOptionModel"` might help, and then just the existence of that test would be sufficient (otherwise unit tests would take infinite time)

Comment (tomsilver) [reply tests/approaches/test_gnn_policy_approach.py:203]: thanks, yeah I saw that, but can you think of an easy way to get the GNN approach to use the mock option model?

Comment (ronuchit) [reply tests/approaches/test_gnn_policy_approach.py:203]: ...it should be clear from the above no? you can see lines like `approach._option_model = _MockOptionModel1(_simulator)  # pylint: disable=protected-access`

Comment (tomsilver) [reply tests/approaches/test_gnn_policy_approach.py]: thx sry was being dum

**Review by ronuchit — APPROVED** [review]
thanks for catching these edge cases! i wonder if this will change our paper results (esp. the "Break early if we have timed out" change)

**Comment by tomsilver** [comment]
if anything, it would make the GNN a bit worse
